pyaqueduct/client/extension_types.py

```diff
@@ -17,6 +17,25 @@ class ExtensionParameterData(BaseModel):
     defaultValue: Optional[str]
     options: Optional[List[str]]
 
+    @classmethod
+    def from_dict(cls, data: dict) -> ExtensionParameterData:
+        """Composes an object from server response
+
+        Args:
+            data: server response
+        
+        Returns:
+            Object populated with server response data
+        """
+        return cls(
+            name=data["name"],
+            displayName=data["displayName"],
+            description=data["description"],
+            dataType=data["dataType"],
+            defaultValue=data["defaultValue"],
+            options=data["options"],
+        )
```

Comment:
I think `BaseModel` has a constructor for this, e.g.:

https://github.com/AqueductHub/aqueductcore/blob/bd3d96edb869af7914b7672e1ef59709b413d21a/aqueductcore/backend/services/extensions.py#L245C25-L245C34c

Reply:
As discussed, the `parse_obj` method from `BaseModel` class from PyDantic cannot be used as that requires the names of all the class variables same as keys in dictionaries passed as parameter.

Hence, we go forward with the `from_dict` method as we are using in our current objects.